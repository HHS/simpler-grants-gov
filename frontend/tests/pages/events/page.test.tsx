import { render, screen, waitFor } from "@testing-library/react";
import { axe } from "jest-axe";
import { identity } from "lodash";
import Events from "src/app/[locale]/events/page";
import {
  localeParams,
  mockMessages,
  useTranslationsMock,
} from "src/utils/testing/intlMocks";

jest.mock("react", () => ({
  ...jest.requireActual<typeof import("react")>("react"),
  use: jest.fn(() => ({
    locale: "en",
  })),
}));

jest.mock("next-intl/server", () => ({
  getTranslations: () => identity,
  setRequestLocale: identity,
}));

jest.mock("next-intl", () => ({
  useTranslations: () => useTranslationsMock(),
  useMessages: () => mockMessages,
}));

describe("Events", () => {
  it("renders Events page unchanged", () => {
    const { container } = render(Events({ params: localeParams }));
    expect(container).toMatchSnapshot();
  });
  it("renders intro text", () => {
    render(Events({ params: localeParams }));

    const content = screen.getByText("pageDescription");

    expect(content).toBeInTheDocument();
  });

  it("passes accessibility scan", async () => {
    const { container } = render(Events({ params: localeParams }));
    const results = await waitFor(() => axe(container));

    expect(results).toHaveNoViolations();
  });
});
