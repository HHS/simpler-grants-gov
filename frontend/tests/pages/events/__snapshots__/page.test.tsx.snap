// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Events renders Events page unchanged 1`] = `
<div>
  <div
    class="events-hero bg-primary-darkest text-white padding-left-6 padding-y-2"
    data-testid="events-hero"
  >
    <div
      class="grid-container"
      data-testid="gridContainer"
    >
      <div
        class="grid-row grid-gap-3 margin-top-6"
        data-testid="grid"
      >
        <div
          class="tablet:grid-col"
          data-testid="grid"
        >
          <h1
            class="tablet:font-sans-2xl desktop-lg:font-sans-3xl desktop-lg:margin-top-2 text-balance"
          >
            header
          </h1>
          <p
            class="usa-intro font-sans-md tablet:font-sans-lg line-height-sans-4 text-balance"
          >
            pageDescription
          </p>
        </div>
        <div
          class="tablet:grid-col"
          data-testid="grid"
        >
          <img
            alt="events-img"
            class="height-auto position-relative padding-top-2 padding-bottom-4"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="grid-container padding-4"
    data-testid="events-upcoming-content"
  >
    <div
      class="grid-row grid-gap-md"
      data-testid="grid"
    >
      <div
        class="tablet:grid-col-3"
        data-testid="grid"
      >
        <h1
          class="margin-left-6"
        >
          title
        </h1>
      </div>
      <div
        class="tablet:grid-col-9 padding-x-6"
        data-testid="grid"
      >
        <span
          class="font-sans-md"
        >
          startDate
        </span>
        <h2>
          header
        </h2>
        <p
          class="font-sans-md line-height-sans-4"
        >
          description
        </p>
        <a
          class="font-sans-md"
          href="https://docs.google.com/forms/d/e/1FAIpQLSe3nyLxAIeky3bGydyvuZobrlEGdWrl0YaZBbVmsn7Pu6HhUw/viewform"
          rel="noopener noreferrer"
          target="_blank"
        >
          link
        </a>
      </div>
    </div>
  </div>
  <div
    class="bg-base-lightest"
  >
    <div
      class="grid-container padding-x-4"
      data-testid="events-demo-content"
    >
      <div
        class="grid-row grid-gap-md padding-6"
        data-testid="grid"
      >
        <div
          class="tablet:grid-col"
          data-testid="grid"
        >
          <img
            alt="events-img"
            class="height-auto position-relative padding-right-6 padding-top-3"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
        </div>
        <div
          class="tablet:grid-col"
          data-testid="grid"
        >
          <div
            class=""
            data-testid="grid"
          >
            <h2>
              title
            </h2>
          </div>
          <div
            class=""
            data-testid="grid"
          >
            <p
              class="font-sans-md line-height-sans-4"
            >
              description
            </p>
            <h3
              class="padding-top-2"
            >
              watch
            </h3>
            <a
              class="font-sans-md line-height-sans-4"
              href="https://vimeo.com/1050177794/278fa78e0b?share=copy"
              rel="noopener noreferrer"
              target="_blank"
            >
              watchLink
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      class="grid-container padding-x-4 bg-base-lightest"
      data-testid="events-coding-content"
    >
      <div
        class="grid-row grid-gap-6 padding-6"
        data-testid="grid"
      >
        <div
          class="tablet:grid-col"
          data-testid="grid"
        >
          <h2>
            title
          </h2>
          <p
            class="font-sans-md line-height-sans-4"
          >
            descriptionP1
          </p>
          <p
            class="font-sans-md line-height-sans-4"
          >
            descriptionP2
          </p>
          <a
            class="font-sans-md line-height-sans-4"
            href="https://wiki.simpler.grants.gov/get-involved/community-events/spring-2025-collaborative-coding-challenge"
            rel="noopener noreferrer"
            target="_blank"
          >
            link
          </a>
        </div>
        <div
          class="tablet:grid-col"
          data-testid="grid"
        >
          <img
            alt="events-img"
            class="height-auto position-relative padding-y-3 padding-left-6"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
